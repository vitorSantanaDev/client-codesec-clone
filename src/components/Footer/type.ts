import { IFooterProps } from 'interfaces/landingPage'

export type IViewProps = IFooterProps
