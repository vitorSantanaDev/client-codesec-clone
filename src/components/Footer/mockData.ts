export const mockDataFooterProps = {
  logo: {
    altText: 'logo codesec',
    urlImage: '/images/logo-2-codesec.svg'
  },
  footerContent: 'Todos od direitos reservados',
  links: [
    {
      href: '/',
      icon: {
        altText: 'icon social',
        urlImage: '/images/instagram.svg'
      }
    },
    {
      href: '/',
      icon: {
        altText: 'icon social',
        urlImage: '/images/facebook.svg'
      }
    },
    {
      href: '/',
      icon: {
        altText: 'icon social',
        urlImage: '/images/youtube.svg'
      }
    }
  ]
}
