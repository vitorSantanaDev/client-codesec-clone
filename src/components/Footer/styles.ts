import styled, { css } from 'styled-components'

export const Wrapper = styled.footer`
  ${({ theme }) => css`
    width: 100%;
    display: flex;
    align-items: center;
    flex-direction: column;
    justify-content: center;

    padding: ${theme.spacings.large} 0;

    .logo {
      width: 134px;
      margin-bottom: ${theme.spacings.small};

      @media ${theme.media.lteMedium} {
        width: 95px;
      }
    }
  `}
`

export const FooterContent = styled.p`
  ${({ theme }) => css`
    color: ${theme.colors.gray7};
    font-size: calc(${theme.font.sizes.small} + 0.2rem);
    margin-bottom: ${theme.spacings.large};

    @media ${theme.media.lteMedium} {
      font-size: ${theme.font.sizes.small};
    }
  `}
`

export const SocialLink = styled.a`
  ${({ theme }) => css`
    margin-right: ${theme.spacings.small};
  `}
`

export const SocialLinkIcon = styled.img`
  ${() => css`
    cursor: pointer;
    transition: ease-in-out all 500ms;

    &:hover {
      transform: translate3d(0px, -7px, 0px);
    }
  `}
`

export const SocialLinksWrapper = styled.div`
  ${({ theme }) => css`
    margin-bottom: ${theme.spacings.small};
  `}
`
