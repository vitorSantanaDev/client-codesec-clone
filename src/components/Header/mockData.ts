import { IViewProps } from './type'

export const mockHeaderProps: IViewProps = {
  logo: {
    urlImage: '/images/codesec-logo.svg',
    altText: 'Logo codesec'
  },
  button: {
    url: '/',
    buttonText: 'Inscreva-se'
  }
}
