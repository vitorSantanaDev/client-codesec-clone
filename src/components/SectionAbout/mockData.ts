export const mockSectionAboutProps = {
  subtitle: 'infos',
  title: 'About web series',
  description:
    'Aprenda PHP do Jeito Certo Utilizando 100% de Tecnologia Moderna e os Melhores Padr√µes de Projeto Nessa Webserie 100% Online e GRATUITA!',
  topics: [
    {
      _id: String(Math.random() * 1000),
      textInfo: 'text 1 example',
      icon: {
        altText: 'icone',
        urlImage: '/images/icon-eye.svg'
      }
    },
    {
      _id: String(Math.random() * 1000),
      textInfo: 'text 1 example',
      icon: {
        altText: 'icone',
        urlImage: '/images/icon-eye.svg'
      }
    },
    {
      _id: String(Math.random() * 1000),
      textInfo: 'text 1 example',
      icon: {
        altText: 'icone',
        urlImage: '/images/icon-eye.svg'
      }
    },
    {
      _id: String(Math.random() * 1000),
      textInfo: 'text 1 example',
      icon: {
        altText: 'icone',
        urlImage: '/images/icon-eye.svg'
      }
    }
  ],
  illustration: {
    urlImage: '/images/about-ilustration.svg',
    altText: 'Illustration about'
  }
}
