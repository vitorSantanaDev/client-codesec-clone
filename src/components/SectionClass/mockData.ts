export const mockDataSectionClassProps = {
  title: 'O que você vai aprender',
  subtitle: 'Aulas',
  classs: [
    {
      dateClass: '14/03',
      hourAndDay: 'Terça as - 20:30H',
      classNumber: 'Aula 01',
      icon: {
        url: 'https://res.cloudinary.com/vsantana/image/upload/v1650500029/icon_lock_6d7fca3f3c.svg',
        altTextIcon: 'Icon'
      },
      titleClass: 'Hackeando tudo',
      descriptionClass:
        'Lorem ipsum dolor sit amet, consectetur adipiscing elit. Est mus nisi, sed in tempor. Nisl sit iaculis nulla consectetur. Purus egestas sit eget viverra. Nisi id in porta nec, a a. Eu in dis et, morbi ut at pellentesque. Massa eu amet morbi eu diam ultricies tellus aliquet. Vel, amet dolor cursus lorem quis amet sed. Mollis diam a purus nulla non, mattis proin odio. Facilisis molestie nibh egestas lacus adipiscing.'
    },
    {
      dateClass: '14/03',
      hourAndDay: 'Terça as - 20:30H',
      classNumber: 'Aula 01',
      icon: {
        url: 'https://res.cloudinary.com/vsantana/image/upload/v1650500029/icon_lock_6d7fca3f3c.svg',
        altTextIcon: 'Icon'
      },
      titleClass: 'Hackeando tudo',
      descriptionClass:
        'Lorem ipsum dolor sit amet, consectetur adipiscing elit. Est mus nisi, sed in tempor. Nisl sit iaculis nulla consectetur. Purus egestas sit eget viverra. Nisi id in porta nec, a a. Eu in dis et, morbi ut at pellentesque. Massa eu amet morbi eu diam ultricies tellus aliquet. Vel, amet dolor cursus lorem quis amet sed. Mollis diam a purus nulla non, mattis proin odio. Facilisis molestie nibh egestas lacus adipiscing.'
    },
    {
      dateClass: '14/03',
      hourAndDay: 'Terça as - 20:30H',
      classNumber: 'Aula 01',
      icon: {
        url: 'https://res.cloudinary.com/vsantana/image/upload/v1650500029/icon_lock_6d7fca3f3c.svg',
        altTextIcon: 'Icon'
      },
      titleClass: 'Hackeando tudo',
      descriptionClass:
        'Lorem ipsum dolor sit amet, consectetur adipiscing elit. Est mus nisi, sed in tempor. Nisl sit iaculis nulla consectetur. Purus egestas sit eget viverra. Nisi id in porta nec, a a. Eu in dis et, morbi ut at pellentesque. Massa eu amet morbi eu diam ultricies tellus aliquet. Vel, amet dolor cursus lorem quis amet sed. Mollis diam a purus nulla non, mattis proin odio. Facilisis molestie nibh egestas lacus adipiscing.'
    },
    {
      dateClass: '14/03',
      hourAndDay: 'Terça as - 20:30H',
      classNumber: 'Aula 01',
      icon: {
        url: 'https://res.cloudinary.com/vsantana/image/upload/v1650500029/icon_lock_6d7fca3f3c.svg',
        altTextIcon: 'Icon'
      },
      titleClass: 'Hackeando tudo',
      descriptionClass:
        'Lorem ipsum dolor sit amet, consectetur adipiscing elit. Est mus nisi, sed in tempor. Nisl sit iaculis nulla consectetur. Purus egestas sit eget viverra. Nisi id in porta nec, a a. Eu in dis et, morbi ut at pellentesque. Massa eu amet morbi eu diam ultricies tellus aliquet. Vel, amet dolor cursus lorem quis amet sed. Mollis diam a purus nulla non, mattis proin odio. Facilisis molestie nibh egestas lacus adipiscing.'
    },
    {
      dateClass: '14/03',
      hourAndDay: 'Terça as - 20:30H',
      classNumber: 'Aula 01',
      icon: {
        url: 'https://res.cloudinary.com/vsantana/image/upload/v1650500029/icon_lock_6d7fca3f3c.svg',
        altTextIcon: 'Icon'
      },
      titleClass: 'Hackeando tudo',
      descriptionClass:
        'Lorem ipsum dolor sit amet, consectetur adipiscing elit. Est mus nisi, sed in tempor. Nisl sit iaculis nulla consectetur. Purus egestas sit eget viverra. Nisi id in porta nec, a a. Eu in dis et, morbi ut at pellentesque. Massa eu amet morbi eu diam ultricies tellus aliquet. Vel, amet dolor cursus lorem quis amet sed. Mollis diam a purus nulla non, mattis proin odio. Facilisis molestie nibh egestas lacus adipiscing.'
    }
  ]
}
