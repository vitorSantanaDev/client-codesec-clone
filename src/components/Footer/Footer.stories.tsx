import { ComponentStory, ComponentMeta } from '@storybook/react'
import Footer from '.'
import { mockDataFooterProps } from './mockData'

export default {
  title: 'Footer',
  component: Footer,
  args: {
    ...mockDataFooterProps
  }
} as ComponentMeta<typeof Footer>

export const Default: ComponentStory<typeof Footer> = (args) => (
  <Footer {...args} />
)
