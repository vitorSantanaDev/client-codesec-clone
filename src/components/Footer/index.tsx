import Logo from 'components/Logo'
import Link from 'next/link'
import * as S from './styles'

type FooterProps = {
  logo: {
    urlImage: string
    altText: string
  }
  footerContent: string
  links: {
    href: string
    icon: {
      urlImage: string
      altText: string
    }
  }[]
}

const Footer = ({ logo, footerContent, links }: FooterProps) => (
  <S.Wrapper>
    <Logo
      className="logo"
      url="/"
      srcImage={logo.urlImage}
      altText={logo.altText}
    />
    <S.FooterContent>{footerContent}</S.FooterContent>
    <S.SocialLinksWrapper>
      {links.map((link, index) => (
        <Link key={`${index}-${link.href}`} href={link.href}>
          <S.SocialLink>
            <S.SocialLinkIcon
              src={link.icon.urlImage}
              alt={link.icon.altText}
            />
          </S.SocialLink>
        </Link>
      ))}
    </S.SocialLinksWrapper>
  </S.Wrapper>
)

export default Footer
