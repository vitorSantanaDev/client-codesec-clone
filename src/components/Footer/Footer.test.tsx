import { screen } from '@testing-library/react'
import { renderTheme } from 'styles/theme/render-theme'
import Footer from '.'
import { mockDataFooterProps } from './mockData'

describe('<Footer />', () => {
  it('should render logo codesec', () => {
    renderTheme(<Footer {...mockDataFooterProps} />)
    const logo = screen.getByAltText(mockDataFooterProps.logo.altText)
    expect(logo).toBeInTheDocument()
  })
  it('should render the footer with correct content', () => {
    renderTheme(<Footer {...mockDataFooterProps} />)
    const content = screen.getByText(mockDataFooterProps.footerContent)
    expect(content).toBeInTheDocument()
  })
  it('should render amount social links correct', () => {
    renderTheme(<Footer {...mockDataFooterProps} />)
    const links = screen.getAllByAltText(
      mockDataFooterProps.socialLinks[0].icon.altText
    )
    expect(links).toHaveLength(3)
  })
  it('should match snapshots', () => {
    const { container } = renderTheme(<Footer {...mockDataFooterProps} />)
    expect(container.firstChild).toMatchSnapshot()
  })
})
