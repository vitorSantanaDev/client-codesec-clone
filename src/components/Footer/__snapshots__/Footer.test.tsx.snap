// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Footer /> should match snapshots 1`] = `
.c1 {
  cursor: pointer;
}

.c0 {
  width: 100%;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  padding: 3.2rem 0;
}

.c0 .logo {
  width: 134px;
  margin-bottom: 1.6rem;
}

.c2 {
  color: #A3A3A3;
  font-size: calc(1.6rem + 0.2rem);
  margin-bottom: 3.2rem;
}

.c4 {
  margin-right: 1.6rem;
}

.c5 {
  cursor: pointer;
  -webkit-transition: ease-in-out all 500ms;
  transition: ease-in-out all 500ms;
}

.c5:hover {
  -webkit-transform: translate3d(0px,-7px,0px);
  -ms-transform: translate3d(0px,-7px,0px);
  transform: translate3d(0px,-7px,0px);
}

.c3 {
  margin-bottom: 1.6rem;
}

@media (max-width:768px) {
  .c0 .logo {
    width: 95px;
  }
}

@media (max-width:768px) {
  .c2 {
    font-size: 1.6rem;
  }
}

<footer
  class="c0"
>
  <div
    class=""
  >
    <img
      alt="logo codesec"
      class="c1 logo"
      src="/images/logo-2-codesec.svg"
    />
  </div>
  <p
    class="c2"
  >
    Todos od direitos reservados
  </p>
  <div
    class="c3"
  >
    <a
      class="c4"
    >
      <img
        alt="icon social"
        class="c5"
        src="/images/instagram.svg"
      />
    </a>
    <a
      class="c4"
    >
      <img
        alt="icon social"
        class="c5"
        src="/images/facebook.svg"
      />
    </a>
    <a
      class="c4"
    >
      <img
        alt="icon social"
        class="c5"
        src="/images/youtube.svg"
      />
    </a>
  </div>
</footer>
`;
